databricks/koalas/frame.py

```diff
@@ -1874,6 +1874,63 @@ def notnull(self):
 
     notna = notnull
 
+    # TODO: add frep and axis parameter
+    def shift(self, periods=1, fill_value=None):
+        """
+        Shift index by desired number of periods.
+
+        When 'freq' is not passed, shift the index without realigning the data.
+        If 'freq' is passed (in this case, the index must be date or datetime, or it will raise
+        a NotImplementedError), the index will be increased using the periods and the 'freq'.
+
+
+        Parameters
+        ----------
+        periods : int
+            Number of periods to shift. Can be positive or negative.
+        fill_value : object, optional
+            The scalar value to use for newly introduced missing values.
+            The default depends on the dtype of self. For numeric data, np.nan is used.
+
+        Returns
+        -------
+        Copy of input object, shifted.
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({'Col1': [10, 20, 15, 30, 45],
+        ...                    'Col2': [13, 23, 18, 33, 48],
+        ...                    'Col3': [17, 27, 22, 37, 52]})
+
+        >>> df.shift(periods=3)
+           Col1  Col2  Col3
+        0   NaN   NaN   NaN
+        1   NaN   NaN   NaN
+        2   NaN   NaN   NaN
+        3  10.0  13.0  17.0
+        4  20.0  23.0  27.0
+
+        >>> df.shift(periods=3, fill_value=0)
+           Col1  Col2  Col3
+        0     0     0     0
+        1     0     0     0
+        2     0     0     0
+        3    10    13    17
+        4    20    23    27
+        """
+        if not isinstance(periods, int):
+            raise ValueError('Periods parameter only support int')
+
+        index = self._internal.index_columns[0]
+        data_columns = self._internal.data_columns
+        sdf = self._sdf
+        window = Window.orderBy(index)
```

Comment:
@RainFung, whenever we use Window, let's add the note https://github.com/databricks/koalas/blob/master/databricks/koalas/frame.py#L5319-L5322